rpc/transport/ws/client.go

```diff
@@ -0,0 +1,88 @@
+package ws
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/rs/zerolog"
+	"github.com/statechannels/go-nitro/rpc/serde"
+	"nhooyr.io/websocket"
+)
+
+type clientWebSocketConnection struct {
+	logger           zerolog.Logger
+	notificationChan chan []byte
+	responseHandlers map[uint64]chan ([]byte)
+	clientWebsocket  *websocket.Conn
+}
+
+// NewWebSocketConnectionAsClient creates a websocket connection that can be used to send requests and listen for notifications
+func NewWebSocketConnectionAsClient(url string) (*clientWebSocketConnection, error) {
+	wsc := &clientWebSocketConnection{}
+	wsc.responseHandlers = make(map[uint64]chan ([]byte))
+	wsc.notificationChan = make(chan []byte)
+
+	conn, _, err := websocket.Dial(context.Background(), url, &websocket.DialOptions{})
+	if err != nil {
+		return nil, err
+	}
+	wsc.clientWebsocket = conn
+	go func() { wsc.readMessages(context.Background()) }()
+	return wsc, nil
+}
+
+func (wsc *clientWebSocketConnection) Request(id uint64, data []byte) ([]byte, error) {
+	// Any request payload type will do here since we are only interested in the id
```

Comment:
Not sure I follow this comment (or the others like it in this file)

Reply:
Agreed that these comments are confusing. We no longer have "any" request/response types. So these comments are removed.